.github/workflows/tests-experimental.yml

```diff
@@ -0,0 +1,70 @@
+name: Tests (experimental)
+
+on:
+  pull_request:
+    paths:
+      # Run only when relevant files are modified
+      - "trl/experimental/**"
+      - "tests/experimental/**"
+
+env:
+  TQDM_DISABLE: 1
+  PYTORCH_CUDA_ALLOC_CONF: "expandable_segments:True"
+  TRL_EXPERIMENTAL_SILENCE: 1
+
+jobs:
+  check_code_quality:
+    name: Check code quality
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.draft == false
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python 3.13
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.13
+      - uses: pre-commit/action@v3.0.1
+        with:
+          extra_args: --all-files
```

Comment:
code quality is already checked in main test.

```suggestion
```

Reply:
Thanks, @qgallouedec.

True: the workflow for regular tests already covers code quality checks.

However, I included it here to handle the case where **only experimental code** is modified. In that scenario, the regular tests (with code quality check) are not triggered. Therefore, the need to check it in this workflow.